kentikapi/client_integration_test.go

```diff
@@ -417,3 +425,110 @@ func timePtr(v time.Time) *time.Time {
 func durationPtr(v time.Duration) *time.Duration {
 	return &v
 }
+
+func TestGRPCClient_GetAgent(t *testing.T) {
+	tests := []struct {
+		name    string
+		id      string
+		res     *pb.GetAgentResponse
+		errCode codes.Code
+		errMsg  string
+	}{
+		{
+			"invalid request with wrong id",
+			"1000",
+			nil,
+			codes.InvalidArgument,
+			"cannot find ID",
+		},
+		{
+			"valid request with valid id",
+			"968",
+			&pb.GetAgentResponse{Agent: newDummyAgentForGRPC()},
+			codes.OK,
+			"",
+		},
+	}
+
+	ctx := context.Background()
+
+	lis, err := net.Listen("tcp", fmt.Sprintf("localhost:%d", port))
+	if err != nil {
+		log.Fatalf("failed to listen: %v", err)
+	}
+	s := grpc.NewServer()
+	pb.RegisterSyntheticsAdminServiceServer(s, &mockSyntheticsServer{})
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			log.Fatalf("failed to serve: %v", err)
+		}
+	}()
+
+	client := kentikapi.NewClient(kentikapi.Config{
+		SyntheticsGRPCAPIURL: lis.Addr().String(),
+	})
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			request := &pb.GetAgentRequest{Id: tt.id}
+
+			response, err := client.GRPCSyntheticsAdminServiceAPI.GetAgent(ctx, request)
+
```

Comment:
Please also verify that requests are properly formed:
- expectedRequestBody
- auth email
- auth token

Please take a look at HTTP service test. It does it all.

Reply:
Done